bytebuffer/bytebufferpool.go

```diff
@@ -0,0 +1,20 @@
+package bytebuffer
+
+import "github.com/valyala/bytebufferpool"
+
+var (
+	// GetLen returns byte buffer with fixed length from the pool, exported from httprouter/bytebuffer.
+	GetLen = func(n int) *bytebufferpool.ByteBuffer {
+		b := bytebufferpool.Get()
```

Comment:
This uses the `defaultPool` of `github.com/valyala/bytebufferpool`. How about allowing the user to set and configure their own pool? They could have several packages that all uses the pool.

Reply:
> This uses the defaultPool of github.com/valyala/bytebufferpool. How about allowing the user to set and configure their own pool?

Actually the bytebufferpool package didn't expose APIs for customizing `Pool` so there is nothing we can do, but I think all configurations in the default pool has been adjusted near perfectly by the author of bytebufferpool.

> They could have several packages that all uses the pool.

I don't think that could be a problem for several packages to share one pool, that is what a pool does, isn't it?